### PR TITLE
Adding Secure URL validation

### PR DESCRIPTION
**Pull request by brumik**
Issue: https://issues.redhat.com/browse/AAH-1195

**Before**
Screenshot from 2022-01-10 14-14-26
Screenshot from 2022-01-10 14-14-15

**After**
Screenshot from 2022-01-20 14-25-28
Screenshot from 2022-01-20 14-24-44
Screenshot from 2022-01-20 14-24-28

**Note**: The fields have the colored underline from the PF when they are not "focused" as you can see them focused in the screenshot.

### CODE REVIEW

File: src/components/repositories/remote-form.tsx [hunk 1]
```diff
@@ -16,12 +16,47 @@ import {
...
+
+  if (url && url.length > 4) {
+    if (url.slice(0, 5) === 'http:') {
+      return {
+        validated: 'warning',
+        helperText: t`Consider using a secure URL (https://).`,
+      };
+    }
+
+    if (url.slice(0, 4) !== 'http') {
```
Comment (himdel) [inline src/components/repositories/remote-form.tsx:49]:
```suggestion
    if (url.slice(0, 6) !== 'https:') {
```

Comment (brumik) [reply src/components/repositories/remote-form.tsx:49]: This change does not make sense for me. My logic without your suggestion:
1. If it is HTTP then show warning (consider using HTTPS)
2. If it is not even starting with HTTP then it cannot be HTTP nor HTTPS so show error that you need a HTTP(s) URL
3. Else the URL is HTTPS so don't show anything

I guess you would like to see this as testing for HTTPS then HTTP then another test for none of these (adds an extra if statement) but since we have 3 error types this makes more sense with returning the "success (default)" with the last return.

Comment (brumik) [reply src/components/repositories/remote-form.tsx:49]: If it helps readibility I can switch up my two `HTTP` and `HTTP:` if cases

Comment (himdel) [reply src/components/repositories/remote-form.tsx:49]: Right, but that's not what the code is doing, `httpie.io` is now a valid value, no warnings ;)

Comment (himdel) [reply src/components/repositories/remote-form.tsx:49]: So.. thinking about this some more, even adding the colon doesn't quite resolve this right, maybe this should be closer to..

```javascript
const httpWarning = {
  validated: 'warning',
  helperText: t`Consider using a secure URL (https://).`,
};

const protocolError = {
  validated: 'error',
  helperTextInvalid: t`The URL needs to be in 'http(s)://' format.`,
};

const success = { validated: 'default' };

try {
  const { protocol } = new URL(url);
  if (protocol === 'https:') {
    return success;
  }
  if (protocol === 'http:') {
    return httpWarning;
  }
} catch (e) {};

return protocolError;
```

File: src/components/repositories/remote-form.tsx [hunk 2]
```diff
@@ -201,16 +236,20 @@ export class RemoteForm extends React.Component<IProps, IState> {
...
+            validated={
+              validateURLHelper(errorMessages['url'], remote.url).validated
+            }
             isRequired={requiredFields.includes('url')}
             isDisabled={disabledFields.includes('url')}
             id='url'
             type='text'
-            value={remote.url || ''}
+            placeholder='https://'
+            value={remote.url}
```
Comment (himdel) [inline src/components/repositories/remote-form.tsx:252]: any reason for this change? All the other form fields are using the `|| ''` pattern here.

File: src/components/repositories/remote-form.tsx (hunk 1 above)
Comment (brumik) [reply src/components/repositories/remote-form.tsx:49]: We are expecting the people to know how URLs are constructed and initially we just wanted to add a `Secure` URL warning. As users probably can use it with IP addresses too and there is no backend check for validity I am starting to think this would be better with two warnings:

`The URL should be in http(s) format` and `Consider using a sercure URL (https)`.  

Also I see why using the `protocol` from the `URL` object is superior, I just kinda did not expected from this change to from a no warning is still mostly working users we go to users which:
1. Will type `httpie.io`
2. They get the warning message about using secure https
3. They get confused :)

I'll update it anyway :)

File: src/components/repositories/remote-form.tsx (hunk 2 above)
Comment (brumik) [reply src/components/repositories/remote-form.tsx]: I think the UI has a really nice way of distingushing the placeholder and current text. I updated it anyway.

File: src/components/repositories/remote-form.tsx [hunk 3]
```diff
@@ -16,12 +16,48 @@ import {
...
+      return {
+        validated: 'warning',
+        helperText: t`Consider using a secure URL (https://).`,
+      };
+    }
+
+    if (protocol === 'https:') {
+      return { validated: 'default' };
+    }
+  } catch (_) {}
```
Comment (himdel) [inline src/components/repositories/remote-form.tsx:53]: Just this lint error, otherwise LGTM :+1: :)

```suggestion
  } catch (_) {
    // fallthrough
  }
```

fixes that :)
